etc.armhf/udev/rules.d/99-com.rules

```diff
@@ -14,39 +14,24 @@ SUBSYSTEM=="gpio", ACTION=="add", PROGRAM="/bin/sh -c 'chgrp -R gpio /sys%p && c
 # PWM export results in a "change" action on the pwmchip device (not "add" of a new device), so match actions other than "remove".
 SUBSYSTEM=="pwm", ACTION!="remove", PROGRAM="/bin/sh -c 'chgrp -R gpio /sys%p && chmod -R g=u /sys%p'"
 
-KERNEL=="ttyAMA0", PROGRAM="/bin/sh -c '\
-	ALIASES=/proc/device-tree/aliases; \
-	if cmp -s $$ALIASES/uart0 $$ALIASES/serial0; then \
-		echo 0;\
-	elif cmp -s $$ALIASES/uart0 $$ALIASES/serial1; then \
-		echo 1; \
-	else \
-		exit 1; \
-	fi\
-'", SYMLINK+="serial%c"
-
-KERNEL=="ttyAMA1", PROGRAM="/bin/sh -c '\
-	ALIASES=/proc/device-tree/aliases; \
-	if [ -e /dev/ttyAMA0 ]; then \
-		exit 1; \
-	elif cmp -s $$ALIASES/uart0 $$ALIASES/serial0; then \
-		echo 0;\
-	elif cmp -s $$ALIASES/uart0 $$ALIASES/serial1; then \
-		echo 1; \
-	else \
-		exit 1; \
-	fi\
-'", SYMLINK+="serial%c"
-
-KERNEL=="ttyS0", PROGRAM="/bin/sh -c '\
-	ALIASES=/proc/device-tree/aliases; \
-	if cmp -s $$ALIASES/uart1 $$ALIASES/serial0; then \
-		echo 0; \
-	elif cmp -s $$ALIASES/uart1 $$ALIASES/serial1; then \
-		echo 1; \
-	else \
-		exit 1; \
-	fi \
+KERNEL=="ttyAMA[0-9]*|ttyS[0-9]*", PROGRAM="/bin/sh -c '\
+        ALIASES=/proc/device-tree/aliases; \
+        TTYNODE=`readlink /sys/class/tty/%k/device/of_node | sed 's/base/:/' | cut -d: -f2`; \
```

Comment:
What about using modern command substitution syntax (same in other cases below)?
```suggestion
        TTYNODE=$(readlink /sys/class/tty/%k/device/of_node | sed 's/base/:/' | cut -d: -f2); \
```
Rationale: https://www.shellcheck.net/wiki/SC2006

Reply:
I'm guessing you haven't, otherwise you'd know it's a bad idea.

Rationale:
```
Jun 19 14:47:39 raspberrypi systemd-udevd[189]: /etc/udev/rules.d/99-com.rules:35 Invalid value "/bin/sh -c 'ALIASES=/proc/device-tree/aliases; TTYNODE=$(readlink /sys/class/tty/%k/device/of_node | sed 's/base/:/' | cut -d: -f2); if [ -e $$ALIASES/bluetooth ] && [ $$TTYNODE/bluetooth = `strings $$ALIASES/bluetooth` ]; then echo 1; elif [ -e $$ALIASES/console ]; then if [ $$TTYNODE = `strings $$ALIASES/console` ]; then echo 0;else exit 1; fi elif [ $$TTYNODE = `strings $$ALIASES/serial0` ]; then echo 0; elif [ $$TTYNODE = `strings $$ALIASES/serial1` ]; then echo 1; else exit 1; fi '" for PROGRAM (char 56: invalid substitution type), ignoring.
```

Reply:
See my last post: Double `$$` needed just like for variable calls.

Reply:
Better?